### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by ShahierNashaat**
Pull request for reviewing connecting to API part using react-redux,

Implement API connecting.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
+};
+
+export const addBook = (payload) => async (dispatch) => {
+  await postBook(payload);
+  dispatch({ type: ADD_BOOK, payload });
+};
+
+export const removeBook = (payload) => async (dispatch) => {
+  await deleteBook(payload);
+  dispatch({ type: REMOVE_BOOK, payload });
```
Comment (SajjadAhmad14) [inline src/redux/books/books.js:34]: Please make `postBook` and `removeBook` server requests with react hooks. All the server requests should be made in the componentDidMount lifecycle method or with react hooks.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,27 +1,47 @@
...
 const initialState = [];
 
-export const addBook = (payload) => ({
-  type: ADD_BOOK,
-  payload,
-});
+const convertDataObjectToArray = (data) => {
+  const dataArray = [];
+
+  Object.keys(data).map((key) => {
```
Comment (SajjadAhmad14) [inline src/redux/books/books.js:12]: The project is unable to run due to the below error. Please make sure the app is running and has no error.
type-error

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Shahier, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/redux/books/books.js (hunk 1 above)
Comment (ShahierNashaat) [reply src/redux/books/books.js:34]: Sorry I can't understand what you mean, could you clarify more?

File: src/api/api.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/KuV7Z2YFTRBTq309Vw2X/books';
+
+export const getAllBooks = () => fetch(URL).then((response) => {
+  if (response.ok) {
+    return response.json();
+  }
+  return {};
+});
```
Comment (SajjadAhmad14) [inline src/api/api.js]: Users after adding books and refreshing the page can see only the first book added in the array. Please make sure all the books are shown on the page.

File: src/api/api.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+export const postBook = (book) => fetch(URL, {
+  method: 'POST',
+  body: JSON.stringify(book),
+  headers: {
+    'Content-Type': 'application/json',
+  },
+}).then((response) => response.text());
+
+export const deleteBook = (id) => fetch(`${URL}/${id}`, {
```
Comment (SajjadAhmad14) [inline src/api/api.js:13]: The user needs to refresh the page to see the book is deleted. Please make sure that the item is removed and disappeared instantly when the user presses the delete button. Kindly check that adding and removing books functionality works fine as before.

File: src/components/Addbook.js [hunk 5]
```diff
@@ -8,36 +8,33 @@ import { addBook } from '../redux/books/books';
 const AddBook = (props) => {
   const { categories } = props;
   const dispatch = useDispatch();
-  const booksLength = useSelector((state) => state.books.length);
+  const booksKeys = useSelector((state) => Math.max(...state.books.map((book) => book.item_id)));
 
   const submitBookToStore = () => {
     const bookTitle = document.querySelector('#bookTitle');
-    const bookAuthor = document.querySelector('#bookAuthor');
```
Comment (SajjadAhmad14) [inline src/components/Addbook.js:20]: [ OPTIONAL ] The book item should have a title and author. Users can only add title when creating a new book for now.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Shahier, 
 ### Thank you for working on changes :ok_hand:

## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by cyonii — APPROVED** [review]
# Approved ✅ 

Hi @ShahierNashaat, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🥳

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**